### DISCUSSION TITLE
fork / clone / imitate? cooking code, and managing sample/snippet content...

### DISCUSSION THREAD

**Discussion by jonnyhotchkiss** [MichaelCurrin/code-cookbook#14] [General]
Hi Michael, I love what you've done with the recipes on the repo... I'm trying to decide how to manage everything, from forum and browser bookmarks and repos and other content from various platforms (google,gist and git) and forums...

(over?)Thinking about a browser extension, assuming it's possible to connect things (ideally syncing with ides like vscode or sonicpi) and login to git|gisthub/google/ to save/sync, and _scraps_ code under cursor... 

Am wondering how you manage your data and workflow, and tips for managing assets, along with any info on how to use this code-cookbook...

Is the answer simply to embrace github|gitlab CI/CD?!

Thanks in advance, captain curious

**Comment by MichaelCurrin** [comment]
Thanks for the message. There's a lot in there but I'll see how I can help.

Aha so I see your aim is to collect and manage snippets that you see as code, answers in forums, and bookmarked URLs. A central place.

I used to use text documents, then EverNote, then Google Keep... but my notes got spread out and I couldn't share them publically easily. And i was concerned about the platforms or tools going away.

So I arrived on my Cheatsheets / Cookbook style of keeping info in a GitHub repo, because it meant it was in the cloud, but not tied to a business. Since I can keep local copy of my Git repo in case GitHub goes down. Or keep backups on GitLab or BitBucket.

Initially it was just markdown files but then I wanted some styling and neat navigation (I had to write my own templating in the Fractal theme using an existing Jekyll theme so I could create the nesting without limit. (Though at the same time, my Cookbook is lacking search functionality so I need to add that with some JS magic and in the meantime I search in VS Code for a line of text like "regex" and look at the Python or JS or grep file).

The benefit then is that the site gets published so I can share links with others, they can contribute, and also I write better quality content not just rough scraps because its my public image. Plus it helps for impressing a potential hirer to show off what you know, and I use it over the years at work too and share with colleagues.

Back to your situation...

Google Keep has a neat labelling system and search ability and didn't have formatting (until recently) so you focus on the content. And it works great as mobile app. It has limitations though such as no hierarchy, and document size limit. So Google Docs (and folders in Google Drive) makes sense for more advanced case. Plus you can share a Google Drive folder as public view-only if you want to share your content with others. But of course there's no CSS theme for that :/ And if you style a page, you have to do that for every page.

I'm a fan of writing Markdown and having the site build with CI/CD. I am guessing you want that to, so the rest below will be for that.

**Comment by MichaelCurrin** [comment]
A good starting point could be forking my repo and deleting all the content. Since a lot of it you won't use or care about and have no personal connection to.
You could also just start with a completely empty repo and add content to it like I have for the recipes directory, and you can add the styling and Jekyll build stuff when you are ready (I don't know you level of using pipelines and Jekyll).

Regarding editing, I like using VS Code locally, or GitHub's VS Code IDE in the browser ( https://github.dev/ since you can move between files without having to commit all the time ).

What you can also try for managing content and previewing styling, is Forestry as a CMS. It won't give you a frontend, but it gives a WordPress-like (or Jekyll Admin-like) way of editing Markdown files in the cloud in a repo you own. And the free tier gives access to a few repos.

You can also go for a private repo. Or one repo for public coding stuff and one for other stuff.

**Comment by MichaelCurrin** [comment]
I'd say figure out the storage location first such as GitHub. And then how you want to add to it like IDE or GitHub editor or Forestry.

I see your aim for making a browser extension so you can select text and send it to your note-taking solution. I don't know if the effort will be worth it though. Just dumping some text somewhere is going get you a mess of pages.

My flow is that I will learn something an anwer to a question or I will find something someone else has done that is impressive, and I'll add it to my notes. If I already have a page for say HTTP requests with Python, I'll add to that existing page I have. Or I'll split the current one into multiple pages. If there's no page already, I'll make one. I'll also look at the surrounding pages if I have time and see how to improve the naming or structure. 

Having a browser extension doesn't make sense then. I am happy just to keep my site open or bookmarked so I can get to it quickly and copy and paste text to it. But I also don't simply paste - I change the format, the wording, I see where on the page it makes sense, I explain where I got it from and where I like it. And the code editor UI we have for that is great - no need to go an reinvent that in a browser extension I think.

**Comment by MichaelCurrin** [comment]
Regarding bookmarks. Well you can have pages in your repo or Google doc where you add a link to a repo and a description about it. 

For a time saver and for saving just the URL and name and category, you can use the Bookmark feature in browsers or a Pocket account with Save to Pocket button in Firefox (probably other browsers too).

**Comment by MichaelCurrin** [comment]
BTW To reduce friction, I added the add button / edit button on my site, so I can jump from the HTML page to the underlying `.md` page. That made me happier and more efficient. So you could enjoy that too if you go with my approach.

**Comment by MichaelCurrin** [comment]
As I said above, you'll want to keep maintaining your garden of notes, not just throwing stuff over the fence and finding its a mess to find anything or there's no structure or some duplication.

To carry on with the metaphore, there's a community of people who maintain their personal "knowledge garden". There's a bunch of tools listed here.

https://github.com/MaggieAppleton/digital-gardeners

Using GitBook might be a great one for you. You get the elegance of a published public website, with content persisted in a repo that it syncs with, but you don't have to maintain any CSS or HTML or JS. On the free tier you only get one site though.

**Comment by MichaelCurrin** [comment]
also you can use chatGPT to help brainstorm ideas with you, using your aims/requirements and asking it for ideas and tradeoffs and tools.

You can also use GPT to format content for you to explain how a piece of code.
